src/Analyzer.Cli.UnitTests/CommandLineParserTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Templates.Analyzer.Cli;
+using System.Linq;
+using System;
+using System.IO;
+
+namespace Analyzer.Cli.FunctionalTests
+{
+    [TestClass]
+    public class CommandLineParserTests
+    {
+        private CommandLineParser _commandLineParser;
+
+        [TestInitialize]
+        public void TestInit()
+        {
+            _commandLineParser = new CommandLineParser();
+        }
+
+        [DataTestMethod]
+        [DataRow("Path does not exist", 2, DisplayName = "Invalid template file path provided")]
+        [DataRow("\\CommandLineParserTests.cs", 4, DisplayName = "Path exists, not an ARM template.")]
+        [DataRow("\\CommandLineParserTests.cs", 3, "--report-format", "Sarif", DisplayName = "Path exists, Report-format flag set, --output-file-path flag not included.")]
+        [DataRow("\\CommandLineParserTests.cs", 1, "--parameters-file-path", DisplayName = "Path exists, Parameters-file-path flag included, but no value provided.")]
+        [DataRow("\\AppServicesLogs-Failures.json", 5, DisplayName = "Issues found in the template")]
+        [DataRow("\\AppServicesLogs-Passes.json", 0, DisplayName = "Success")]
+        public void AnalyzeTemplate_ValidInputValues_ReturnExpectedExitCode(string relativeTemplatePath, int expectedExitCode, params string[] additionalParams)
```

Comment:
Rename `additionalParams` to something more descriptive, like `cliOptions` or `additionalCliOptions`.  Same with other tests.